OpenKh.Tests/Patcher/PatcherTests.cs

```diff
@@ -514,6 +514,638 @@ public void MergeKh2AreaDataScriptTest()
             });
         }
 
+        [Fact]
+        public void ListReplaceTrsrTest()
+        {
+            var patcher = new PatcherProcessor();
+            var serializer = new Serializer();
+            var patch = new Metadata() { 
+                Assets = new List<AssetFile>()
+                {
+                    new AssetFile()
+                    {
+                        Name = "03system.bar",
+                        Method = "binarc",
+                        Source = new List<AssetFile>()
+                        {
+                            new AssetFile()
+                            {
+                                Name = "trsr",
+                                Method = "listreplace",
+                                Type = "List",
+                                Source = new List<AssetFile>()
+                                {
+                                    new AssetFile()
+                                    {
+                                        Name = "TrsrList.yml",
+                                        Type = "trsr"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            File.Create(Path.Combine(AssetsInputDir, "03system.bar")).Using(stream =>
+            {
+                var trsrEntry = new List<Kh2.SystemData.Trsr>()
+                {
+                    new Kh2.SystemData.Trsr
+                    {
+                        Id = 1,
+                        ItemId = 10
+                    }
+                    };
+                using var trsrStream = new MemoryStream();
+                Kh2.SystemData.Trsr.Write(trsrStream, trsrEntry);
+                Bar.Write(stream, new Bar() {
+                    new Bar.Entry()
+                    {
+                        Name = "trsr",
+                        Type = Bar.EntryType.List,
+                        Stream = trsrStream
+                    }
+                });
+            });
+
+            File.Create(Path.Combine(ModInputDir, "TrsrList.yml")).Using(stream =>
+            {
+                var writer = new StreamWriter(stream);
+                writer.WriteLine("1:");
+                writer.WriteLine("  Id: 1");
```

Comment:
Is the key always equal to the Id? If that is the case it might be possible to simplify this.